core/foundation/src/FoundationUtils.cxx

```diff
@@ -113,6 +123,23 @@ const std::string& GetFallbackRootSys() {
       };
       fallback = parent_path(parent_path(lpFilename));
    }
+#elif defined __FreeBSD__
+  procstat* ps = procstat_open_sysctl();  //
+  kinfo_proc* kp = kinfo_getproc(getpid());
+
+  if (kp!=NULL) {
+     char path_str[PATH_MAX] = "";
+     procstat_getpathname(ps, kp, path_str, sizeof(path_str));
+     std::string execpath(path_str);
+     
+      auto parent_path = [](std::string path) {
+         return path.substr(0, path.find_last_of("/\\"));
+      };
+      fallback = parent_path(parent_path({path_str}));
```

Comment:
I see that this part is common with the `WIN32` case.  Perhaps we can restructure this a bit to avoid code duplication.

Reply:
possibly, I even tried, but for some reason this was better if I remember correctly... maybe not a good reson though:)